app/controllers/concerns/idv_session.rb

```diff
@@ -12,12 +12,36 @@ def confirm_idv_session_started
     redirect_to idv_session_url unless idv_session[:params].present?
   end
 
+  def confirm_idv_attempts_allowed
+    if idv_attempts_exceeded?
+      flash[:error] = t('idv.errors.hardfail')
+      redirect_to idv_fail_url
+    end
+  end
+
   def idv_session
     user_session[:idv] ||= {}
   end
 
   protected
 
+  def idv_attempts=(num)
+    user_session[:idv_attempts] = num
+  end
+
+  def idv_attempts
+    user_session[:idv_attempts] ||= 0
+  end
```

Comment:
Usage like this:
```ruby
self.idv_attempts += 1
```
requires both a getter and a setter. That example line is from the review controller.